src/hbb/processors/GenSelection.py

```diff
@@ -0,0 +1,251 @@
+    fatjets["HiggsMatchIndex"] = ak.mask(
+        ak.argmin(fatjets.metric_table(higgs), axis=2), fatjets["HiggsMatch"] == 1
+    )
+    fatjets["NumBMatchedH1"] = ak.sum(fatjets.metric_table(b_h1) < 0.8, axis=2)
+    fatjets["NumBMatchedH2"] = ak.sum(fatjets.metric_table(b_h2) < 0.8, axis=2)
+    fatjets["MaxdRH1"] = ak.max(fatjets.metric_table(b_h1), axis=2)
+    fatjets["MaxdRH2"] = ak.max(fatjets.metric_table(b_h2), axis=2)
+
+    num_fatjets = 2
+    bbFatJetVars = {
+        f"bbFatJet{var}": pad_val(fatjets[var], num_fatjets, axis=1)
+        for var in [
+            "HiggsMatch",
+            "HiggsMatchIndex",
+            "NumBMatchedH1",
+            "NumBMatchedH2",
+            "MaxdRH1",
+            "MaxdRH2",
+        ]
+    }
+
+    return {**GenHiggsVars, **GenbVars, **ak4JetVars, **bbFatJetVars}
+
+
+def gen_selection_Hbb(
+    events: NanoEventsArray,
+    jets: JetArray,  # noqa: ARG001
+    fatjets: FatJetArray,
+    selection_args: list,  # noqa: ARG001
+    skim_vars: dict,
+):
+    """Gets H, bb, 4-vectors + Higgs children information"""
+
+    # finding the two gen higgs
+    higgs = events.GenPart[
+        (abs(events.GenPart.pdgId) == HIGGS_PDGID) * events.GenPart.hasFlags(GEN_FLAGS)
+    ]
+    higgs_children = higgs.children
+
+    GenHiggsVars = {f"GenHiggs{key}": higgs[var].to_numpy() for (var, key) in skim_vars.items()}
+    GenHiggsVars["GenHiggsChildren"] = abs(higgs_children.pdgId[:, :, 0]).to_numpy()
+
+    is_bb = abs(higgs_children.pdgId) == b_PDGID
+    bs = ak.flatten(higgs_children[is_bb], axis=2)
+    GenbVars = {f"Genb{key}": pad_val(bs[var], 4, axis=1) for (var, key) in skim_vars.items()}
+
+    # match fatjets to bb
+    # bs_unflat = higgs_children[is_bb]
+    # num_b_matched = ak.sum(fatjets.metric_table(bs_unflat) < 0.8, axis=2)
+    #[:, n] selects the  n -th element along the second axis (the Higgs boson index)
+    b_h1 = ak.firsts(higgs_children[is_bb][:, 0:1])   # first b quark
+    b_h2 = ak.firsts(higgs_children[is_bb][:, 1:2]) # second b quark
+    matched_to_higgs = fatjets.metric_table(higgs) < 0.8    # metric_table returns the deltaR between the fatjet and the higgs
+    is_fatjet_matched = ak.any(matched_to_higgs, axis=2)
+
+    fatjets["HiggsMatch"] = is_fatjet_matched
+    fatjets["HiggsMatchIndex"] = ak.mask(
+        ak.argmin(fatjets.metric_table(higgs), axis=2), fatjets["HiggsMatch"] == 1
+    )
+    fatjets["NumBMatchedH1"] = ak.sum(fatjets.metric_table(b_h1) < 0.8, axis=2)
+    fatjets["NumBMatchedH2"] = ak.sum(fatjets.metric_table(b_h2) < 0.8, axis=2)
+
+    print(ak.num(fatjets[var], axis=1))
+
+    num_fatjets = 2
+    bbFatJetVars = {
+        f"bbFatJet{var}": pad_val(fatjets[var], num_fatjets, axis=1)
+        for var in [
+            "HiggsMatch",
+            "HiggsMatchIndex",
+            "NumBMatchedH1",
+            "NumBMatchedH2",
+        ]
+    }
+
+    return {**GenHiggsVars, **GenbVars, **bbFatJetVars}
+
+
+def gen_selection_V(
```

Comment:
clean up this function and add comments to what the variables "VMatch"  and "NumMatched" mean